app/src/main/java/com/hedvig/app/feature/crossselling/ui/detail/CrossSellDetailViewModel.kt

```diff
@@ -2,45 +2,66 @@ package com.hedvig.app.feature.crossselling.ui.detail
 
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import arrow.core.Either
+import com.hedvig.app.feature.crossselling.model.NavigateChat
+import com.hedvig.app.feature.crossselling.model.NavigateEmbark
 import com.hedvig.app.feature.crossselling.ui.CrossSellData
 import com.hedvig.app.feature.embark.quotecart.CreateQuoteCartUseCase
-import com.hedvig.app.feature.home.ui.changeaddress.appendQuoteCartId
 import com.hedvig.hanalytics.HAnalytics
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 
 class CrossSellDetailViewModel(
-    private val crossSell: CrossSellData,
+    typeOfContract: String,
+    private val crossSellAction: CrossSellData.Action,
     hAnalytics: HAnalytics,
     private val createQuoteCartUseCase: CreateQuoteCartUseCase,
 ) : ViewModel() {
-    private val _action = MutableStateFlow<CrossSellData.Action?>(null)
-    val action = _action.asStateFlow()
+
+    private val _viewState = MutableStateFlow(ViewState())
+    val viewState = _viewState.asStateFlow()
 
     init {
-        hAnalytics.screenViewCrossSellDetail(crossSell.typeOfContract)
+        hAnalytics.screenViewCrossSellDetail(typeOfContract)
     }
 
+    data class ViewState(
+        val navigateEmbark: NavigateEmbark? = null,
+        val navigateChat: NavigateChat? = null,
+        val errorMessage: String? = null,
+        val loading: Boolean = false,
+    )
```

Comment:
Since we are passing `CrossSellData` as an extra for this activity - I think it's ok to not get it from the view model state and use it to build the UI directly in the activity. I see the view model state as something that can change depending on interaction with the view model, the `CrossSellData` is static in this case.